### DISCUSSION TITLE
Boost timestep in IterationAdaptiveDT?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#23145] [Q&A General] [answered]
In `IterationAdaptiveDT`, there are lots of ways to limit the timestep, but is there a way to _boost_ it? For instance, if I know, maybe based on a postprocessor value, that I'm in a period that is nearly steady state, can I bump up my timestep? I know the timestep will increase eventually, but I may want to increase it by several orders of magnitude but that can take a while even with a modest growth factor.

**Comment by GiudGiud** [comment, marked as answer]
Well yes, there is a growth_factor parameter but it doesnt work with postprocessors. There s a limiting option with postprocessors only. 


We need some sort of CombinedTimestepper with a reduction operation such as min/max/avg
that will let us select the timestep based on multiple timesteppers.

It seems here you know of a region where a functionDT could have large steps

**Reply by jessecarterMOOSE** [reply]
If I reformulate my problem, I can know the time(s) where I can take bigger steps. It looks like [this example](https://github.com/idaholab/moose/blob/next/test/tests/time_steppers/iteration_adaptive/adapt_tstep_grow_dtfunc.i) shows how to do this using the `time_t` and `time_dt` parameters to `IterationAdaptiveDT` (though the [documentation](https://mooseframework.inl.gov/source/timesteppers/IterationAdaptiveDT.html) doesn't say much about how this feature works).

**Reply by hugary1995** [reply]
> We need some sort of CombinedTimestepper with a reduction operation such as min/max/avg
> that will let us select the timestep based on multiple timesteppers.

That is one possible design to improve the flexibility of the TimeStepper system. An alternative is to refactor the TimeStepper system as TimeStepModifiers, similar to MeshModifiers in that they can be represented as a directed acyclic graph, where we can do the usual dependency resolution business.

A lot of things could benefit from DAGs, the MeshModifier and Executor are good examples... I hope someday we can compose the _entire_ MOOSE simulation as a big DAG....
